packages/frontend/app/nav/subnav.ts

```diff
@@ -0,0 +1,85 @@
+import { makeNavLink } from './navigation';
+
+const flatOrParentSubnav = (
```

Comment:
Ditto to above - I wonder if we should pass just enough in the frontend model to avoid repeating this logic here? As again, it's pretty complex/a lot of it.